wpimath/src/main/java/edu/wpi/first/math/controller/ElevatorFeedforward.java

```diff
@@ -99,76 +123,150 @@ public double getKv() {
    * @return The computed feedforward.
    */
-  public double calculate(double currentVelocity, double nextVelocity, double dtSeconds) {
-    // Discretize the affine model.
+  public Voltage calculate(LinearVelocity currentVelocity, LinearVelocity nextVelocity) {
+    // For an elevator with the model
+    // dx/dt = −kᵥ/kₐ x + 1/kₐ u - kg/kₐ - kₛ/kₐ sgn(x),
     //
-    //   dx/dt = Ax + Bu + c
-    //   dx/dt = Ax + B(u + B⁺c)
-    //   xₖ₊₁ = eᴬᵀxₖ + A⁻¹(eᴬᵀ - I)B(uₖ + B⁺cₖ)
-    //   xₖ₊₁ = A_d xₖ + B_d (uₖ + B⁺cₖ)
-    //   xₖ₊₁ = A_d xₖ + B_duₖ + B_d B⁺cₖ
+    // where
+    // A = −kᵥ/kₐ
+    // B = 1/kₐ
+    // c = -(kg/kₐ + kₛ/kₐ sgn(x))
+    // A_d = eᴬᵀ
+    // B_d = A⁻¹(eᴬᵀ - I)B
+    // dx/dt = Ax + Bu + c
     //
-    // Solve for uₖ.
-    //
-    //   B_duₖ = xₖ₊₁ − A_d xₖ − B_d B⁺cₖ
-    //   uₖ = B_d⁺(xₖ₊₁ − A_d xₖ − B_d B⁺cₖ)
-    //   uₖ = B_d⁺(xₖ₊₁ − A_d xₖ) − B⁺cₖ
+    // Discretize the affine model.
+    // dx/dt = Ax + Bu + c
+    // dx/dt = Ax + B(u + B⁺c)
+    // xₖ₊₁ = eᴬᵀxₖ + A⁻¹(eᴬᵀ - I)B(uₖ + B⁺cₖ)
+    // xₖ₊₁ = A_d xₖ + B_d (uₖ + B⁺cₖ)
+    // xₖ₊₁ = A_d xₖ + B_d uₖ + B_d B⁺cₖ
     //
-    // For an elevator with the model
-    // dx/dt = -Kv/Ka x + 1/Ka u - Kg/Ka - Ks/Ka sgn(x),
-    // A = -Kv/Ka, B = 1/Ka, and c = -(Kg/Ka + Ks/Ka sgn(x)). Substitute in B
-    // assuming sgn(x) is a constant for the duration of the step.
+    // Solve for uₖ.
+    // B_d uₖ = xₖ₊₁ − A_d xₖ − B_d B⁺cₖ
+    // uₖ = B_d⁺(xₖ₊₁ − A_d xₖ − B_d B⁺cₖ)
+    // uₖ = B_d⁺(xₖ₊₁ − A_d xₖ) − B⁺cₖ
     //
-    //   uₖ = B_d⁺(xₖ₊₁ − A_d xₖ) − Ka(-(Kg/Ka + Ks/Ka sgn(x)))
-    //   uₖ = B_d⁺(xₖ₊₁ − A_d xₖ) + Ka(Kg/Ka + Ks/Ka sgn(x))
-    //   uₖ = B_d⁺(xₖ₊₁ − A_d xₖ) + Kg + Ks sgn(x)
-    var plant = LinearSystemId.identifyVelocitySystem(this.kv, this.ka);
-    var feedforward = new LinearPlantInversionFeedforward<>(plant, dtSeconds);
-
-    var r = MatBuilder.fill(Nat.N1(), Nat.N1(), currentVelocity);
-    var nextR = MatBuilder.fill(Nat.N1(), Nat.N1(), nextVelocity);
-
-    return kg + ks * Math.signum(currentVelocity) + feedforward.calculate(r, nextR).get(0, 0);
-  }
-
-  /**
-   * Calculates the feedforward from the gains and velocity setpoint (acceleration is assumed to be
-   * zero).
-   *
-   * @param velocity The velocity setpoint.
-   * @return The computed feedforward.
-   */
-  public double calculate(double velocity) {
-    return calculate(velocity, 0);
+    // Substitute in B assuming sgn(x) is a constant for the duration of the step.
+    // uₖ = B_d⁺(xₖ₊₁ − A_d xₖ) − kₐ(-(kg/kₐ + kₛ/kₐ sgn(x)))
+    // uₖ = B_d⁺(xₖ₊₁ − A_d xₖ) + kₐ(kg/kₐ + kₛ/kₐ sgn(x))
+    // uₖ = B_d⁺(xₖ₊₁ − A_d xₖ) + kg + kₛ sgn(x)
+    if (ka == 0.0) {
+      // Simplify the model when kₐ = 0.
+      //
+      // Simplify A.
+      // A = −kᵥ/kₐ
+      // As kₐ approaches zero, A approaches -∞.
+      // A = −∞
+      //
+      // Simplify A_d.
+      //
+      // A_d = eᴬᵀ
+      // A_d = exp(−∞)
+      // A_d = 0
+      //
+      // Simplify B_d.
```

Comment:
Add newlines between text and math blocks, and indent math blocks by two spaces.